### PR TITLE
Execute the tests on pull request

### PR DESCRIPTION
**Pull request by Letme**

### CODE REVIEW

**Comment by JasperCraeghs** [comment]
The workflow is executed twice now. I doubt this is what we want. https://github.com/melexis/sphinx-traceability-extension/pull/291/checks

File: .github/workflows/python-package.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 name: Test and deploy on tag
 
-on: push
+on: [push, pull_request]
```
Comment (Letme) [inline .github/workflows/python-package.yml:3]:
```suggestion
on:
  tags:
  push:
    branches:
    - master
  pull_request:
```

File: .github/workflows/python-package.yml [hunk 2]
```diff
@@ -1,6 +1,11 @@
 name: Test and deploy on tag
 
-on: push
+on:
+  tags:
+  push:
+    branches:
+    - master
+  pull_request:
```
Comment (Letme) [inline .github/workflows/python-package.yml:8]:
```suggestion
on:
  push:
    tags:
    branches:
    - master
  pull_request:
  pull_request_target:
    types: [assigned, opened, synchronize, reopened]
```

File: .github/workflows/python-package.yml [hunk 3]
```diff
@@ -1,6 +1,12 @@
 name: Test and deploy on tag
 
-on: push
+on:
+  push:
+    tags:
+    branches:
+    - master
+  pull_request_target:
+    types: [assigned, opened, synchronize, reopened]
```
Comment (Letme) [inline .github/workflows/python-package.yml:9]:
```suggestion
on:
  push:
    tags:
    branches:
    - master
  pull_request:
    types: [assigned, opened, synchronize, reopened]
  pull_request_target:
    types: [assigned, opened, synchronize, reopened]
```

**Comment by Letme** [comment]
OK, this seems to be the combination we want (also need push_target for the forks). I hope the tags work without having:

```
tag:
  - *
```

**Pull request merged**